connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConfigTest.java

```diff
@@ -56,6 +57,39 @@
             "set X-Frame-Options:DENY, add  :no-cache, no-store, must-revalidate "
     );
 
+    @Test
+    public void testListenersConfigAllowedValues() {
+        Map<String, String> props = baseProps();
+
+        // no value set for "listeners"
+        WorkerConfig config = new WorkerConfig(WorkerConfig.baseConfigDef(), props);
+        assertEquals(LISTENERS_DEFAULT, config.getList(WorkerConfig.LISTENERS_CONFIG));
+
+        props.put(WorkerConfig.LISTENERS_CONFIG, "http://a.b:9999");
+        config = new WorkerConfig(WorkerConfig.baseConfigDef(), props);
+        assertEquals(config.getList(WorkerConfig.LISTENERS_CONFIG), Arrays.asList("http://a.b:9999"));
+
+        props.put(WorkerConfig.LISTENERS_CONFIG, "http://a.b:9999, https://a.b:7812");
+        config = new WorkerConfig(WorkerConfig.baseConfigDef(), props);
+        assertEquals(config.getList(WorkerConfig.LISTENERS_CONFIG), Arrays.asList("http://a.b:9999", "https://a.b:7812"));
+
+        new WorkerConfig(WorkerConfig.baseConfigDef(), props);
+    }
+
+    @Test
+    public void testListenersConfigNotAllowedValues() {
+        Map<String, String> props = baseProps();
+
+        props.put(WorkerConfig.LISTENERS_CONFIG, "");
+        assertThrows(ConfigException.class, () -> new WorkerConfig(WorkerConfig.baseConfigDef(), props)).printStackTrace();
```

Comment:
Did you consider adding another check for a list with all blank values`,,,`? It may not be handled any differently than the next check, so maybe its not worth it. The benefit is that it _might_ help as a test against future regressions if the logic is changed.

Reply:
Very good suggestion, I added that case.